### PR TITLE
Feature - Update Flow Login Without Logout And A Unique User

### PR DESCRIPTION
**Pull request by at-hangtran**

### CODE REVIEW

File: src/test/java/stepdefs/AuthenticateDefinitions.java [hunk 1]
```diff
@@ -1,6 +1,11 @@
 package stepdefs;
 
 import cucumber.api.java8.En;
+import org.openqa.selenium.By;
+import org.openqa.selenium.NotFoundException;
+import org.openqa.selenium.TimeoutException;
```
Comment (at-vinhhuynh) [inline src/test/java/stepdefs/AuthenticateDefinitions.java]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/akaj-team/selenium-test/blob/996e15635766fdc6d38b3f25b4466557c99471b9/src/test/java/stepdefs/AuthenticateDefinitions.java#L6"></span>Unused import - org.openqa.selenium.TimeoutException.</td>
    </tr>
  </tbody>
</table>

File: src/test/java/stepdefs/AuthenticateDefinitions.java [hunk 2]
```diff
@@ -1,6 +1,11 @@
...
+import org.openqa.selenium.support.ui.ExpectedConditions;
```
Comment (at-vinhhuynh) [inline src/test/java/stepdefs/AuthenticateDefinitions.java]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/akaj-team/selenium-test/blob/996e15635766fdc6d38b3f25b4466557c99471b9/src/test/java/stepdefs/AuthenticateDefinitions.java#L7"></span>Unused import - org.openqa.selenium.support.ui.ExpectedConditions.</td>
    </tr>
  </tbody>
</table>

File: src/test/java/stepdefs/AuthenticateDefinitions.java [hunk 3]
```diff
@@ -1,6 +1,11 @@
...
+import org.openqa.selenium.support.ui.WebDriverWait;
```
Comment (at-vinhhuynh) [inline src/test/java/stepdefs/AuthenticateDefinitions.java]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/akaj-team/selenium-test/blob/996e15635766fdc6d38b3f25b4466557c99471b9/src/test/java/stepdefs/AuthenticateDefinitions.java#L8"></span>Unused import - org.openqa.selenium.support.ui.WebDriverWait.</td>
    </tr>
  </tbody>
</table>

File: src/test/java/stepdefs/AuthenticateDefinitions.java [hunk 4]
```diff
@@ -18,38 +23,41 @@
     public AuthenticateDefinitions() {
         initMap();
         Given("^I am logged in as (a|an) \"([^\"]*)\"$", (String arg0, String position) -> {
-            clearCookies();
-
-            if (!getDriver().getCurrentUrl().startsWith("data")) {
-                getDriver().executeScript("window.localStorage.clear();");
+            if(!isButtonLogoutDisplayed()){
```
Comment (at-vinhhuynh) [inline src/test/java/stepdefs/AuthenticateDefinitions.java]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/akaj-team/selenium-test/blob/996e15635766fdc6d38b3f25b4466557c99471b9/src/test/java/stepdefs/AuthenticateDefinitions.java#L26"></span>‘if’ is not followed by whitespace.</td>
    </tr>
  </tbody>
</table>

Comment (at-vinhhuynh) [inline src/test/java/stepdefs/AuthenticateDefinitions.java]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/akaj-team/selenium-test/blob/996e15635766fdc6d38b3f25b4466557c99471b9/src/test/java/stepdefs/AuthenticateDefinitions.java#L26"></span>’{‘ is not preceded with whitespace.</td>
    </tr>
  </tbody>
</table>

File: src/test/java/stepdefs/HomeDefinitions.java [hunk 5]
```diff
@@ -28,7 +30,12 @@ public HomeDefinitions() {
         homePage = initPage(getDriver(), HomePage.class);
 
         // Check Home page is started
-        Given("^I am stayed in home page$", () -> waitForPageDisplayed(driver, URL_HOME_PAGE, By.className("wapper")));
+        Given("^I am stayed in home page$", () -> {
+            getDriver().get(HOME_PAGE_URL);
+            new WebDriverWait(getDriver(), 10).until(
```
Comment (at-vinhhuynh) [inline src/test/java/stepdefs/HomeDefinitions.java]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/akaj-team/selenium-test/blob/996e15635766fdc6d38b3f25b4466557c99471b9/src/test/java/stepdefs/HomeDefinitions.java#L35"></span>‘10’ is a magic number.</td>
    </tr>
  </tbody>
</table>

File: src/test/java/vn/asiantech/page/LoginPage.java [hunk 6]
```diff
@@ -32,40 +35,38 @@ public LoginPage login() {
...
 
-    public WebElement getLoginButton() {
+    public final WebElement getLoginButton() {
         return loginButton;
     }
 
-    public LoginPage waitForErrorMessage(WebDriver driver, int timeOutInSecond) {
-        waitForElement(driver, errorText, timeOutInSecond);
-        return this;
+    public final void waitForErrorMessage(final WebDriver driver) {
```
Comment (at-vinhhuynh) [inline src/test/java/vn/asiantech/page/LoginPage.java]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/akaj-team/selenium-test/blob/996e15635766fdc6d38b3f25b4466557c99471b9/src/test/java/vn/asiantech/page/LoginPage.java#L68"></span>‘driver’ hides a field.</td>
    </tr>
  </tbody>
</table>

File: src/test/java/stepdefs/AuthenticateDefinitions.java [hunk 7]
```diff
@@ -18,38 +20,41 @@
     public AuthenticateDefinitions() {
         initMap();
         Given("^I am logged in as (a|an) \"([^\"]*)\"$", (String arg0, String position) -> {
-            clearCookies();
-
-            if (!getDriver().getCurrentUrl().startsWith("data")) {
-                getDriver().executeScript("window.localStorage.clear();");
+            if (!isButtonLogoutDisplayed()) {
+                Account account = accounts.get(position);
+                getDriver().get("http://portal-stg.asiantech.vn/auth/login");
```
Comment (at-tienhoang) [inline src/test/java/stepdefs/AuthenticateDefinitions.java]: Move the url to Constant class

File: src/test/java/stepdefs/AuthenticateDefinitions.java [hunk 8]
```diff
@@ -18,38 +20,41 @@
...
+                loginPage = initPage(getDriver(), LoginPage.class);
```
Comment (at-tienhoang) [inline src/test/java/stepdefs/AuthenticateDefinitions.java]: Move this line to first line of constructor

**Pull request merged**